x-pack/legacy/plugins/actions/server/builtin_action_types/lib/result_type.ts

```diff
@@ -42,6 +42,13 @@ export function isErr<T, E>(result: Result<T, E>): result is Err<E> {
   return !isOk(result);
 }
 
+export function unsafeGet<T, E>(result: Result<T, E>): T {
+  if (isErr(result)) {
+    throw new Error(`${result.error}`);
```

Comment:
I don't think we want to create a new error here - or I don't see why we need to, and so then we probably shouldn't.  We'd lose the stack trace from the original error.  So, seems like it should just be `throw result.error`.

Reply:
It's never meant to be used (hence unsafe), but it lets you keep the tests cleaner.
I could extract this into the tests if that helps.

Reply:
I could make the change you suggest, but didn't originally because "E" isn't necessarily an Error type... it could be a primitive for example.
I see your point about stack trace though... I might do something like "check if it's an error, if not, create an error, if yes, throw as is"

Reply:
Oh wait, maybe `result.error` isn't an error, but a string?  In which case you'd use `throw new Error(result.error)`; but I guess we really don't know WHAT the type of result.error is?  That makes this expression confusing to me - what is the actual type of `result.error` here?  Could we type it in `result_type.ts` as an `Error` instead of as a parameterized type?

Reply:
haha yeah, exactly :)
A Result doesn't have to return an Error type on purpose - a failed operation doesn't necessarily mean an error occurred (despite the name).

For now I think I'll just extract it to the tests since it shouldn't be used in real implementation.